### PR TITLE
module relevancy condition fix, test

### PR DESCRIPTION
**Pull request by wpride**
Fix for http://manage.dimagi.com/default.asp?244868

We were evaluating relevancy conditions even when the menu wasn't going to be displayed because its root wasn't the current screen.

### CODE REVIEW

**Comment by benrudolph** [comment]
test failure looks real, but otherwise looks good on ✔️ . nice find

**Comment by wpride** [comment]
@benrudolph This evolved a little bit. I implemented a real fix yesterday on commcare-core that centralized the menu navigation logic. The error from the test was real so I resolved by merging in the fix I made last night. That fix also included some error handling that uses platform-wide localized strings, so I added those the Formplayer as well. Then I added the nepali and ethiopian month lists to those as well since Jeremy had that bug where they were missing

The TLDR is we can add strings to the formplayer_translatable_strings.txt file which will be localizable across Formplayer now

File: src/main/resources/formplayer_translatable_strings.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+app.menu.display.cond.bad.xpath=Invalid form display condition (${0}): ${1}
+app.menu.display.cond.xpath.err=Error evaluating form display condition (${0}): ${1}
+ethiopian_months=Mäskäräm,T’ïk’ïmt,Hïdar,Tahsas,T’ïr,Yäkatit,Mägabit,Miyaziya,Gïnbot,Säne,Hämle,Nähäse,P’agume
+nepali_months=Baishakh,Jestha,Ashadh,Shrawan,Bhadra,Ashwin,Kartik,Mangsir,Poush,Magh,Falgun,Chaitra
```
Comment (benrudolph) [inline src/main/resources/formplayer_translatable_strings.txt]: oo interesting
